src/hooks/insecure-parser.js

```diff
@@ -0,0 +1,12 @@
+/* eslint-disable no-underscore-dangle */
+
+const insecureParserHook = (options) => {
+    if (options.context.insecureParser !== undefined) {
+        options._unixOptions = {
```

Comment:
This is quite hacky but Got currently doesn't support custom native options...

Reply:
And we can't add it, because it's an `Options` instance and it would throw... 😄 Would be good to add this comment to the code.

Reply:
`_unixOptions` is an internal that exists on `optionsInstance._unixOptions` so we *can* use this 😅

Reply:
Yeah, I know. But we're using an internal property so it's nice to comment it. E.g. like this:

```js
// insecureHTTPParser is not an allowed HTTP setting of Got so we must
// inject it directly into the Got internals for Node.js HTTP module to use it.
```

Reply:
Perfect